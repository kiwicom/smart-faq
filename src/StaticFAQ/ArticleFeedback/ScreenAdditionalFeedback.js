// @flow

import * as React from 'react';
import css from 'styled-jsx/css';
import { Heading, Text, Button } from '@kiwicom/orbit-components';
import { Close } from '@kiwicom/orbit-components/lib/icons';

import { Box } from '../../common';
import createComment from '../../mutations/CreateCommentMutation';
import screensList from './screensList';

type Props = {|
  changeScreen: (nextScreen: string) => void,
  articleId: string,
  commentType: string,
  clearCommentType: () => void,
|};

type State = {|
  comment: string,
|};

const style = css`
  div.question {
    margin-top: 16px;
    margin-bottom: 4px;
  }
  div.inputArea.invalid textarea {
    border-color: #d21c1c;
  }
  div.inputArea p {
    font-family: 'Roboto';
    font-size: 12px;
    font-weight: 400;
    color: #d21c1c;
  }
  div.inputArea textarea {
    width: 100%;
    height: 72px;
    border-radius: 3px;
    background-color: #ffffff;
    border: solid 1px #bac7d5;
    resize: none;
  }
  div.inputArea textarea::placeholder {
    color: #bac7d5;
  }
  div.button {
    margin-top: 19px;
  }
  div.close-icon {
    position: absolute;
    top: 8px;
    right: 8px;
    cursor: pointer;
  }
  div.inputArea textarea {
    padding: 12px 16px;
    font-size: 14px;
  }
  @media only screen and (max-width: 901px) {
    div.inputArea textarea {
      width: 100%;
    }
  }
`;

class ScreenAdditionalFeedback extends React.Component<Props, State> {
  state = {
    comment: '',
  };

  handleChange = (e: SyntheticInputEvent<HTMLInputElement>) => {
    this.setState({ comment: e.target.value });
  };

  handleSubmit = (e: SyntheticEvent<HTMLFormElement>) => {
    e.preventDefault();
    const { changeScreen, articleId, commentType } = this.props;
    const { comment } = this.state;

    createComment(
      articleId,
      commentType,
      comment,
      () => changeScreen(screensList.THANK_YOU),
      () => changeScreen(screensList.ERROR),
    );
    this.props.clearCommentType();
  };

  closeScreen = () => {
    this.props.changeScreen(screensList.VOTING);
    this.props.clearCommentType();
  };

  render() {
    return (
      <Box
        border="none"
        padding="40px 24px 24px 24px"
        borderRadius="4px"
        backgroundColor="#f5f7f9"
      >
        <form onSubmit={this.handleSubmit}>
          <div
            className="close-icon"
            onClick={this.closeScreen}
            onKeyUp={null}
            tabIndex="-1"
            role="button"
          >
            <Close customColor="#bac7d5" size="small" />
          </div>
          <Heading type="title3">
            Please tell us more about your issue with the article (optional)
          </Heading>
          <div className="question">
            <Text>Comment</Text>
          </div>
          <div className="inputArea">
            <textarea
              data-gramm_editor="false"
              onChange={this.handleChange}
              value={this.state.comment}
              placeholder="How could this be better?"
            />
          </div>
          <div className="button">
            <Button onClick={() => {}}>Submit</Button>
          </div>
        </form>
        <style jsx>{style}</style>
      </Box>
    );
  }
}

export default ScreenAdditionalFeedback;
