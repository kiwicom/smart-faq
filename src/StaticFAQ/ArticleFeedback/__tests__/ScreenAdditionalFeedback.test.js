// @flow

import * as React from 'react';
import { shallow } from 'enzyme';

import ScreenAdditionalFeedback from '../ScreenAdditionalFeedback';

const ch = (sc: string) => {}; //eslint-disable-line

describe('ScreenAdditionalFeedback', () => {
  const component = shallow(
    <ScreenAdditionalFeedback
      changeScreen={ch}
      articleId="WE4rT"
      commentType="OTHER"
      clearCommentType={() => {}}
    />,
  );

  it('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
