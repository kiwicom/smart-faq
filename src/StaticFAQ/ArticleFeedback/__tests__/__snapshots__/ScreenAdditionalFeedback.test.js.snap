// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenAdditionalFeedback should match snapshot 1`] = `
<Box
  backgroundColor="#f5f7f9"
  border="none"
  borderRadius="4px"
  padding="40px 24px 24px 24px"
>
  <form
    className="jsx-2730678398"
    onSubmit={[Function]}
  >
    <div
      className="jsx-2730678398 close-icon"
      onClick={[Function]}
      onKeyUp={null}
      role="button"
      tabIndex="-1"
    >
      <Close
        customColor="#bac7d5"
        size="small"
      />
    </div>
    <WithTheme(Heading)
      type="title3"
    >
      Please tell us more about your issue with the article (optional)
    </WithTheme(Heading)>
    <div
      className="jsx-2730678398 question"
    >
      <WithTheme(Text)>
        Comment
      </WithTheme(Text)>
    </div>
    <div
      className="jsx-2730678398 inputArea"
    >
      <textarea
        className="jsx-2730678398"
        data-gramm_editor="false"
        onChange={[Function]}
        placeholder="How could this be better?"
        value=""
      />
    </div>
    <div
      className="jsx-2730678398 button"
    >
      <WithTheme(Button)
        onClick={[Function]}
      >
        Submit
      </WithTheme(Button)>
    </div>
  </form>
  <JSXStyle
    css="div.question.jsx-2730678398{margin-top:16px;margin-bottom:4px;}div.inputArea.invalid.jsx-2730678398 textarea.jsx-2730678398{border-color:#d21c1c;}div.inputArea.jsx-2730678398 p.jsx-2730678398{font-family:'Roboto';font-size:12px;font-weight:400;color:#d21c1c;}div.inputArea.jsx-2730678398 textarea.jsx-2730678398{width:100%;height:72px;border-radius:3px;background-color:#ffffff;border:solid 1px #bac7d5;resize:none;}div.inputArea.jsx-2730678398 textarea.jsx-2730678398::-webkit-input-placeholder{color:#bac7d5;}div.inputArea.jsx-2730678398 textarea.jsx-2730678398::-moz-placeholder{color:#bac7d5;}div.inputArea.jsx-2730678398 textarea.jsx-2730678398:-ms-input-placeholder{color:#bac7d5;}div.inputArea.jsx-2730678398 textarea.jsx-2730678398::placeholder{color:#bac7d5;}div.button.jsx-2730678398{margin-top:19px;}div.close-icon.jsx-2730678398{position:absolute;top:8px;right:8px;cursor:pointer;}div.inputArea.jsx-2730678398 textarea.jsx-2730678398{padding:12px 16px;font-size:14px;}@media only screen and (max-width:901px){div.inputArea.jsx-2730678398 textarea.jsx-2730678398{width:100%;}}"
    styleId="2730678398"
  />
</Box>
`;
